db/db_batch.go

```diff
@@ -170,16 +162,24 @@ func (b *baseKVStoreBatch) clear() error {
 // NewCachedBatch returns a new cached batch buffer
 func NewCachedBatch() CachedBatch {
 	return &cachedBatch{
-		cache: make(map[hash.PKHash][]byte),
+		cache: make(map[hash.CacheHash][]byte),
 	}
 }
 
+// ClearAndUnlock clears the write queue and unlocks the batch
+func (cb *cachedBatch) ClearAndUnlock() {
+	defer cb.mutex.Unlock()
+	cb.cache = nil
```

Comment:
I don't think = nil is necessary. Garbage collection will handle the unreferenced memory. cc @yutongp